### PR TITLE
feat: create OakPrimaryNavItem and OakPrimaryNav components

### PR DESCRIPTION
**Pull request by weronika-szalas**
Create `OakPrimaryNavItem` and `OakPrimaryNav` components

`OakPrimaryNav`:
https://deploy-preview-92--lively-meringue-8ebd43.netlify.app/?path=/docs/components-integrated-oakprimarynav--docs

`OakPrimaryNavItem`:
https://deploy-preview-92--lively-meringue-8ebd43.netlify.app/?path=/docs/components-integrated-oakprimarynavitem--docs

Mobile view of this navigation will be a separate component and it will be controlled with `display: none` or `display: block` to display relevant version depending on the viewport.

Screenshot 2024-02-01 at 14 40 30

### CODE REVIEW

File: src/components/integrated/OakPrimaryNav/OakPrimaryNav.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export const OakPrimaryNav = ({ ariaLabel, navItems }: OakPrimaryNavProps) => {
+  return (
+    <nav aria-label={ariaLabel}>
+      <OakFlex as={"ul"}>
+        {navItems.map(
+          ({ href, isCurrent, label, ...rest }: OakPrimaryNavItemProps) => (
+            <OakLI
+              $listStyle={"none"}
+              $mr={["space-between-m", "space-between-m", "space-between-none"]}
```
Comment (kimon-satan) [inline src/components/integrated/OakPrimaryNav/OakPrimaryNav.tsx:23]: this doesn't seem to make sense . We end up with larger margins on the side and the bottom for the mobile view whilst none for desktop ? Also we end up with some unnecessary margin on the rhs after the final element this way. Could we instead use the $gap property of parent flex and also the flex direction ?

**Review by kimon-satan — CHANGES_REQUESTED** [review]
Just one comment about the margins

File: src/components/integrated/OakPrimaryNav/OakPrimaryNav.tsx (hunk 1 above)
Comment (weronika-szalas) [reply src/components/integrated/OakPrimaryNav/OakPrimaryNav.tsx]: updated

File: src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      {label}
+    </OakPrimaryButton>
+  ) : (
+    <OakPrimaryInvertedButton
+      element={Link}
+      href={href}
+      shallow={shallow}
+      isTrailingIcon={true}
+      aria-label={label}
+      aria-disabled={isCurrent}
```
[the remark this answers is not in the document]
Comment (weronika-szalas) [reply src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx:58]: yes, I wanted to discuss this with you - it's either above or we can make it a span element

File: src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    >
+      {label}
+    </OakPrimaryButton>
+  ) : (
+    <OakPrimaryInvertedButton
+      element={Link}
+      href={href}
+      shallow={shallow}
+      isTrailingIcon={true}
+      aria-label={label}
```
[the remark this answers is not in the document]
Comment (weronika-szalas) [reply src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx]: updated

File: src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    >
+      {label}
+    </OakPrimaryButtonAriaDisabled>
+  ) : (
+    <OakPrimaryInvertedButton
+      element={Link}
+      href={href}
+      shallow={shallow}
+      isTrailingIcon={true}
+      aria-disabled={isCurrent}
```
Comment (rightsaidjames) [inline src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx:58]: Is this attribute no longer needed now that we're handling the disabled state without ARIA? Its value can now only be false, as far as I can tell.

File: src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+export const OakPrimaryNavItem = ({
+  href,
+  isCurrent = false,
+  shallow = true,
+  label,
+  ...rest
+}: OakPrimaryNavItemProps) => {
+  return isCurrent ? (
+    <OakPrimaryButtonAriaDisabled
```
Comment (rightsaidjames) [inline src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx:45]: `Aria` is no longer being used to make the link inactive, so does the component need renaming?

File: src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx (hunk 4 above)
Comment (weronika-szalas) [reply src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx]: removed

File: src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx (hunk 5 above)
Comment (weronika-szalas) [reply src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx]: renamed

**Review by rightsaidjames — APPROVED** [review]
Lookin' good

File: src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  OakPrimaryInvertedButtonProps,
+} from "@/components/ui/OakPrimaryInvertedButton";
+
+type Url = string | UrlObject;
+
+export type OakPrimaryNavItemProps = {
+  href: Url;
+  isCurrent?: boolean;
+  shallow?: boolean;
+  label: string;
```
Comment (carlmw) [inline src/components/integrated/OakPrimaryNavItem/OakPrimaryNavItem.tsx:22]: minor, but this should probably be `children` and the type `ReactNode` to fit into the components it's wrapping. As a consumer i'd expect the API to be the same as a regular button/link.

**Review by carlmw — APPROVED** [review]
No blockers for me just a minor. As discussed in our call I reckon this component can be decomposed a little later for re-use elsewhere 👍🏼 :shipit: .

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**